### DISCUSSION TITLE
[@types/geojson] has correct `Position` type commented out

### DISCUSSION THREAD

**Discussion by thw0rted** [DefinitelyTyped/DefinitelyTyped#58642] [Issues with a @types package]
See https://github.com/Turfjs/turf/issues/1906

Position should be a tuple so that `pos[0]` is not undefined under `nouncheckedindexedaccess` (i.e. in Strict mode).  I'm not sure whether it should be `[number, number, number?]` or `[number, number, number?, number?]`, to allow for M-values.  I lean towards the latter, because I can't think of a way it would break for those who don't need/use/allow M-values.

[a comment by an automated account is not shown]

**Comment by icholy** [comment]
Your problem is that `[number, number, number?]` isn't correctly typed. `[number, number, number?]` is not assignable to `number[]` because `undefined` is not assignable to `number`.

Change your function parameter type to be a union and it will work:

`[number, number] | [number, number, number] | [number, number, number, number]`.

**Reply by thw0rted** [reply]
Ah right, makes sense, but still it should definitely not just be `number[]`.  Does it make sense to replace the current type with the tuple-union?

**Reply by icholy** [reply]
I wasn't the one that made the change from tuple-union to `number[]`. But I'm assuming it was done because the tuple union is annoying to use. You can assign a `[number, number]` to `number[]`, but not the other way around. Changing it back to a tuple-union would break a lot of code.

**Reply by thw0rted** [reply]
Oh that's fun: https://github.com/DefinitelyTyped/DefinitelyTyped/commit/f12705b83b35eb0c9c96aaf6211113431738022a#diff-6e65366e2fe1c28603e289bb653ba87e9a8bf71a154a05bb2bfb519e6643efeaL37

This appears to be the last commit where the line was changed.  No reason was given, 3 people showed up to say "why was changed?", no answer.  It looks like the author of that commit hasn't been active here on DT in a couple of years.  Sounds like it would make sense to change it back (adding the four-tuple option).

**Reply by icholy** [reply]
No it wouldn't make sense. Like I said, it would break a tone of code.

**Reply by thw0rted** [reply]
I think that breaking *incorrect* code should be OK, especially when this can catch errors by forcing people to check the number of elements in an array.  It also avoids spurious bounds checks when you use strict mode.

That said, I think you get most of the benefits from changing to `[number, number, ...number[]]`.  I just tried and that syntax guarantees `pos[0]` and `pos[1]` are defined while still being assign-able to `number[]`.  Would that work?

**Reply by JeffJacobson** [reply]
> Oh that's fun: [f12705b#diff-6e65366e2fe1c28603e289bb653ba87e9a8bf71a154a05bb2bfb519e6643efeaL37](https://github.com/DefinitelyTyped/DefinitelyTyped/commit/f12705b83b35eb0c9c96aaf6211113431738022a#diff-6e65366e2fe1c28603e289bb653ba87e9a8bf71a154a05bb2bfb519e6643efeaL37)
> 
> This appears to be the last commit where the line was changed. No reason was given, 3 people showed up to say "why was changed?", no answer. It looks like the author of that commit hasn't been active here on DT in a couple of years. Sounds like it would make sense to change it back (adding the four-tuple option).

I haven't been able to work on this GeoJSON type def for a long time. Please feel free to make whatever changes you need to without my approval.

IIRC, the last time I modified the type definition was after the official GeoJSON specs were updated. Before my involvement the Position type was a `number[]`. I changed it to be `[number, number] | [number, number, number]` as part of that pull request. It was requested that I change it back before the pull request would be approved, so I did.

**Reply by thw0rted** [reply]
I'm having trouble finding the PR, but based on the adjacent comment, I wonder if the issue was that there was disagreement over whether the four-tuple should be allowed or not.

I think it's worth noting that when you were asked to change it back to an unbounded array, that was before `nouncheckedindexedaccess` existed, so there wasn't much advantage to having a tuple of N identical types vs an array of that type.  It didn't matter then, but it matters now.

**Reply by JeffJacobson** [reply]
Is there anything I need to do to turn over "ownership" of this type definition? (I'm not even the person who originally created this one, so I don't consider myself the "owner" anyway.) 

I don't want to be a bottleneck and have people waiting for me to sign off on something.

**Reply by thw0rted** [reply]
I think the bot treats anyone listed in the header as an "owner", so taking yourself out of that block should stop you from being notified.  (I think anybody on the list can approve a PR, so being on the list doesn't slow down the process or anything.)
